### PR TITLE
JBTM-3754 Allow the JDBCStore to be configured directly from a DataSo…

### PR DESCRIPTION
**Pull request by mmusgrov**
https://issues.redhat.com/browse/JBTM-3754

CORE

### CODE REVIEW

**Comment by marcosgopen** [comment]
Thanks @mmusgrov for your PR. I started the CI test for it.

**Comment by mmusgrov** [comment]
Thanks Marco.

File: ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/objectstore/JDBCStoreDSTest.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+import com.arjuna.ats.internal.arjuna.objectstore.jdbc.JDBCStore;
+import com.arjuna.common.internal.util.propertyservice.BeanPopulator;
+import org.h2.jdbcx.JdbcDataSource;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.sql.SQLException;
+
+import static org.junit.Assert.*;
```
Comment (jmfinelli) [inline ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/objectstore/JDBCStoreDSTest.java:24]: Do you think it's possible to use JUnit 5?

File: ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/objectstore/jdbc/JDBCStore.java [hunk 2]
```diff
@@ -201,65 +204,79 @@ public JDBCStore(ObjectStoreEnvironmentBean jdbcStoreEnvironmentBean) throws Obj
...
+            } else if (connectionDetails != null) {
                 StringTokenizer stringTokenizer = new StringTokenizer(connectionDetails, ";");
-                JDBCAccess jdbcAccess = (JDBCAccess) Class.forName(stringTokenizer.nextToken()).newInstance();
+                jdbcAccess = (JDBCAccess) Class.forName(stringTokenizer.nextToken()).newInstance();
                 jdbcAccess.initialise(stringTokenizer);
+            } else {
+                throw new ObjectStoreException("Invalid store configuration");
+            }
 
-                _storeName = jdbcAccess.getClass().getName() + ":" + tableName;
```
Comment (jmfinelli) [inline ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/objectstore/jdbc/JDBCStore.java:214]: Could you please group all formatting changes in a separate commit? Thanks :-)

**Review by jmfinelli — COMMENTED** [review]
Thanks @mmusgrov! I left a couple of comments

File: ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/objectstore/JDBCStoreDSTest.java (hunk 1 above)
Comment (mmusgrov) [reply ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/objectstore/JDBCStoreDSTest.java:24]: I was keeping it in line with the other tests in this package but you are correct, we've got to start somewhere.

File: ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/objectstore/jdbc/JDBCStore.java (hunk 2 above)
Comment (mmusgrov) [reply ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/objectstore/jdbc/JDBCStore.java:214]: This one is not reformatting - I moved this code into its own method: `private void initImple(...`.

**Review by jmfinelli — APPROVED** [review]
LGTM

**Pull request merged**